plugin/pkg/shoot/resourcereservation/apis/shootresourcereservation/types.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shootresourcereservation
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Configuration provides configuration for the ShootResourceReservation admission controller.
+type Configuration struct {
+	metav1.TypeMeta
+	// UseGKEFormula enables the calculation of resource reservations based on
+	// the CPU and memory resources available for a machine type.
+	UseGKEFormula bool `json:"useGKEFormula,omitempty"`
```

Comment:
Why did you add `omitempty` here?

Reply:
No particular reason. The configuration fields in other admission plugins with configuration also seem to use `omitempty`, see for example `plugin/pkg/shoot/tolerationrestriction/apis/shoottolerationrestriction/v1alpha1/types.go`. So, that's essentially copy&paste from there. But I can also drop the `omitempty` if you want.

Reply:
Typically, `omitempty` is only used for optional fields. If this is an optional field, you should change the type to `*bool` as well and add the `// +optional` tag to the comment. In this case, I think it's not optional, so let's remove the `omitempty` then :)

Reply:
Ok, I've removed the omitempty in both places.